link/internal/services/machine.go

```diff
@@ -66,6 +68,35 @@ func CreateMachine(data []byte, log *slog.Logger, url string) ([]byte, error) {
 	return responseData, nil
 }
 
+func CreateMachineGRPC(c *client.Client, data []byte, log *slog.Logger, url string) ([]byte, error) {
```

Comment:
why do yo have data with []byte type? If it is a grpc server it should be possible to generate internal models. The same for rest service. 
[]byte - is a generic, but your schema could be defined in the code